File: src/Corretora.Api/Controllers/Sbp/NewSbpEventRequest.cs
namespace Corretora.Api.Controllers.Sbp
{
    public class NewSbpEventRequest
    {
        public string Event { get; set; }
        public TargetRequest Target { get; set; }
        public OriginRequest Origin { get; set; }
        public decimal Amount { get; set; }
    }

    public class TargetRequest
    {
        public string Bank { get; set; }
        public string Branch { get; set; }
        public string Account { get; set; }
    }

    public class OriginRequest
    {
        public string Bank { get; set; }
        public string Branch { get; set; }
        public string Cpf { get; set; }
    }
}

File: src/Web/angular/src/app/services/new-sbp-event.request.ts
export interface NewSbpEventRequest {
  event: string;
  target: TargetDto;
  origin: OriginDto;
  amount: number;
}

export interface TargetDto {
  bank: string;
  branch: string;
  account: string;
}

export interface OriginDto {
  bank: string;
  branch: string;
  cpf: string;
}
